File: CS/DXGrid_AssignComboBoxToColumn/MainViewModel.cs
using System.Collections.Generic;
using System.Linq;

namespace DXGrid_AssignComboBoxToColumn {
    public class Person {
        public string ProductName { get; set; }
        public int Type { get; set; }
        public int UnitPrice { get; set; }

        public Person(int i) {
            ProductName = "FirstName" + i;
            UnitPrice = i;
            Type = 0;
        }
    }

    public class ProductType {
        public int Id { get; set; }
        public string TypeName { get; set; }

        public ProductType(int i) {
            Id = i;
            TypeName = "Type" + i;
        }
    }

    public class MyViewModel {
        public List<Person> PersonList { get; set; }
        public List<ProductType> TypeList { get; set; }

        public MyViewModel() {
            PersonList = new List<Person>(GetPersonData(20));
            TypeList = new List<ProductType>(GetTypeData(3));
        }

        static IEnumerable<Person> GetPersonData(int count) {
            return Enumerable.Range(0, count).Select(x => new Person(x));
        }

        static IEnumerable<ProductType> GetTypeData(int count) {
            return Enumerable.Range(0, count).Select(x => new ProductType(x));
        }
    }
}

File: VB/DXGrid_AssignComboBoxToColumn/MainViewModel.vb
Imports System.Collections.Generic
Imports System.Linq

Namespace DXGrid_AssignComboBoxToColumn

    Public Class Person

        Public Property ProductName As String

        Public Property Type As Integer

        Public Property UnitPrice As Integer

        Public Sub New(ByVal i As Integer)
            ProductName = "FirstName" & i
            UnitPrice = i
            Type = 0
        End Sub
    End Class

    Public Class ProductType

        Public Property Id As Integer

        Public Property TypeName As String

        Public Sub New(ByVal i As Integer)
            Id = i
            TypeName = "Type" & i
        End Sub
    End Class

    Public Class MyViewModel

        Public Property PersonList As List(Of Person)

        Public Property TypeList As List(Of ProductType)

        Public Sub New()
            PersonList = New List(Of Person)(GetPersonData(20))
            TypeList = New List(Of ProductType)(GetTypeData(3))
        End Sub

        Private Shared Function GetPersonData(ByVal count As Integer) As IEnumerable(Of Person)
            Return Enumerable.Range(0, count).[Select](Function(x) New Person(x))
        End Function

        Private Shared Function GetTypeData(ByVal count As Integer) As IEnumerable(Of ProductType)
            Return Enumerable.Range(0, count).[Select](Function(x) New ProductType(x))
        End Function
    End Class
End Namespace
